docs/source/_src_user_guide/state_propagation/environment_setup/architecture/custom_settings.rst

```diff
@@ -0,0 +1,121 @@
+
+Custom environment models:
+
+* Custom aerodynamic coefficients (with suppported dependencies from :class:`~tudatpy.numerical_simulation.environment.AerodynamicCoefficientsIndependentVariables`) :func:`~tudatpy.numerical_simulation.environment_setup.aerodynamic_coefficients.custom_aerodynamic_force_coefficients` or :func:`~tudatpy.numerical_simulation.environment_setup.aerodynamic_coefficients.custom_aerodynamic_force_and_moment_coefficients` (only force, and force and moment coefficients, respectively)
+* Custom atmospheric density model (function of time only) :func:`~tudatpy.numerical_simulation.environment_setup.atmosphere.custom_constant_temperature`
+* Custom atmospheric density model (function of position and time) :func:`~tudatpy.numerical_simulation.environment_setup.atmosphere.custom_four_dimensional_constant_temperature` *Note:* this custom function has current time, altitude, latitude and longitude as input.
+* Custom wind model: :func:`~tudatpy.numerical_simulation.environment_setup.atmosphere.custom_wind_model`
+* Custom ephemeris: :func:`~tudatpy.numerical_simulation.environment_setup.ephemeris.custom_ephemeris`
+
+Custom propagation models:
+
+* Custom body orientation :func:`~tudatpy.numerical_simulation.propagation_setup.thrust.custom_thrust_orientation`
+* Custom body direction :func:`~tudatpy.numerical_simulation.propagation_setup.thrust.custom_thrust_direction`
+* Custom thrust magnitude :func:`~tudatpy.numerical_simulation.propagation_setup.thrust.custom_thrust_direction`
+* Custom torque :func:`~tudatpy.numerical_simulation.propagation_setup.torque.custom_torque`
+* Custom acceleration :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.custom_acceleration`
+* Custom mass rate :func:`~tudatpy.numerical_simulation.propagation_setup.mass_rate.custom_mass_rate`
+* Custom state :func:`~tudatpy.numerical_simulation.propagation_setup.propagation.custom_state` *Note:* this custom function has current time, and the current custom state, as input
+* Custom termination setting :func:`~tudatpy.numerical_simulation.propagation_setup.propagator.custom_termination`  *Note:* this custom function takes no inputs, if the current time is needed, it can be retrieved from the :attr:`~tudatpy.numerical_simulation.environment.SystemOfBodies.current_time` attribute of the :class:`~tudatpy.numerical_simulation.environment.SystemOfBodies`
+* Custom dependent variable :func:`~tudatpy.numerical_simulation.propagation_setup.dependent_variable.custom_dependent_variable` *Note:* this custom function takes no inputs, if the current time is needed, it can be retrieved from the :attr:`~tudatpy.numerical_simulation.environment.SystemOfBodies.current_time` attribute of the :class:`~tudatpy.numerical_simulation.environment.SystemOfBodies`
+
```

Comment:
Small question/suggestion: would it be worth adding one example for creating a custom termination setting and/or dependent variable? They are a bit different from the other custom "models" since time is not (necessarily) an input. Actually, one might argue that termination settings/dependent variables are not really "models" and they could be treated independently from the rest of the custom propagation models?